File: kotlin/refactoring-du-bouchonnois/src/test/kotlin/bouchonnois/tests/doubles/PartieDeChasseRepositoryForTests.kt
package bouchonnois.tests.doubles

import bouchonnois.domain.PartieDeChasse
import bouchonnois.repository.PartieDeChasseRepository
import java.util.*

class PartieDeChasseRepositoryForTests : PartieDeChasseRepository {
    var savedPartieDeChasse: PartieDeChasse? = null
    private val partiesDeChasse: Map<UUID, PartieDeChasse> = mutableMapOf()

    override fun save(partieDeChasse: PartieDeChasse) {
        savedPartieDeChasse = partieDeChasse
        partiesDeChasse + (partieDeChasse.id to partieDeChasse)
    }

    override fun getById(partieDeChasseId: UUID?): PartieDeChasse? =
        partiesDeChasse.getOrDefault(partieDeChasseId, null)
}

File: c#/Bouchonnois.Tests/Doubles/PartieDeChasseRepositoryForTests.cs
using Bouchonnois.Domain;
using Bouchonnois.Repository;

namespace Bouchonnois.Tests.Doubles
{
    public class PartieDeChasseRepositoryForTests : IPartieDeChasseRepository
    {
        private readonly IDictionary<Guid, PartieDeChasse> _partiesDeChasse = new Dictionary<Guid, PartieDeChasse>();
        private PartieDeChasse _savedPartieDeChasse;

        public void Save(PartieDeChasse partieDeChasse)
        {
            _savedPartieDeChasse = partieDeChasse;
            _partiesDeChasse[partieDeChasse.Id] = partieDeChasse;
        }

        public PartieDeChasse GetById(Guid partieDeChasseId)
            => (_partiesDeChasse.ContainsKey(partieDeChasseId)
                ? _partiesDeChasse[partieDeChasseId]
                : null)!;

        public void Add(PartieDeChasse partieDeChasse) => _partiesDeChasse[partieDeChasse.Id] = partieDeChasse;
        public PartieDeChasse SavedPartieDeChasse() => _savedPartieDeChasse;
    }
}